pyodide_http/_streaming.py

```diff
@@ -0,0 +1,231 @@
+        // magic number for header ready
+        Atomics.store(intBuffer, 0, SUCCESS_HEADER);
+        Atomics.notify(intBuffer,0);
+        Atomics.wait(intBuffer, 0, intBuffer[0], 500);// wait until it resets to zero which means read
+        const reader = response.body.getReader();
+        while (true) {
+            let { value, done } = await reader.read();
+            if (value) {
+                // pass a chunk back to the sharedarraybuffer - wait half a second or else assume it
+                // didn't get consumed for some reason and abort
+                let curOffset = 0;
+                while (curOffset < value.length) {
+
+                    let curLen = value.length - curOffset;
+                    if (curLen > byteBuffer.length) {
+                        curLen = byteBuffer.length;
+                    }
+                    byteBuffer.set(value.subarray(curOffset, curOffset+curLen), 0)
+                    Atomics.store(intBuffer, 0, curLen);// store current length in bytes
+                    Atomics.notify(intBuffer,0);
+                    Atomics.wait(intBuffer, 0, intBuffer[0]);// wait until it resets to zero 
+                    if(intBuffer[0]!=0)
+                    {
+                        throw "Error - intBuffer not read";
+                    }
+                    // which means it is read by the other end
+                    curOffset += curLen;
+                }
+            }
+            if (done) {
+                Atomics.store(intBuffer, 0, SUCCESS_EOF);
+                Atomics.notify(intBuffer,0);
+                // finished reading successfully
+                // return from event handler 
+                return;
+            }
+        }
+    }
+    catch (error) {
+            console.log("Request exception:",error);
+            let errorBytes=encoder.encode(errorBytes);
+            let written = errorBytes.length;
+            byteBuffer.set(errorBytes);
+            intBuffer[1] = written;
+            Atomics.store(intBuffer, 0, ERROR_EXCEPTION);
+            Atomics.notify(intBuffer,0);
+        }
+    }
+
+);
+
+"""
+
+
+def _obj_from_dict(dict_val: dict) -> any:
+    return to_js(dict_val, dict_converter=js.Object.fromEntries)
+
+
+class _ReadStream(io.RawIOBase):
+    def __init__(self, int_buffer, byte_buffer):
+        self.int_buffer = int_buffer
+        self.byte_buffer = byte_buffer
+        self.read_pos = 0
+        self.read_len = 0
+
+    def readable(self) -> bool:
+        return True
+
+    def writeable(self) -> bool:
+        return False
+
+    def seekable(self) -> bool:
+        return False
+
+    def readinto(self, byte_obj) -> bool:
+        if not self.int_buffer:
+            return 0
+        if self.read_len == 0:
+            # wait for the worker to send something
+            js.Atomics.store(self.int_buffer, 0, 0)
+            js.Atomics.notify(self.int_buffer, 0)
+            js.Atomics.wait(self.int_buffer, 0, 0, 5000)
```

Comment:
What happens when the `5000` milliseconds is passed? Maybe let the end-user specify the timeout? Or we can for now just create a new ticket so we don't forget about this.

Reply:
I added support for requests timeout parameter  for this and the other timeout